src/main/java/net/sf/jsqlparser/parser/ParserKeywordsUtils.java

```diff
@@ -0,0 +1,429 @@
+                , { "RS", RESTRICTED_JSQLPARSER }
+                , { "UR", RESTRICTED_JSQLPARSER }
+                , { "CS", RESTRICTED_JSQLPARSER }
+
+                //@todo: Object Names should not start with Hex-Prefix, we shall not find that Token
+                , { "0x", RESTRICTED_JSQLPARSER }
+        };
+
+        ArrayList<String> keywords = new ArrayList<>();
+        for (Object[] data : ALL_RESERVED_KEYWORDS) {
+            int value = (int) data[1];
+
+            // test if bit is not set
+            if ( (value & restriction) == restriction
+                    || (restriction & value) == value ) {
+                keywords.add((String) data[0]);
+            }
+        }
+
+        return keywords;
+    }
+
+    public static void main(String[] args) throws Exception {
+        if (args.length<1) {
+            throw new IllegalArgumentException("No filename provided as parameters ARGS[0]");
+        }
+
+        File file = new File(args[0]);
+        if (file.exists() && file.canRead()) {
+            buildGrammarForRelObjectName(file);
+            buildGrammarForRelObjectNameWithoutValue(file);
+        } else {
+            throw new FileNotFoundException("Can't read file " + args[0]);
+        }
+    }
+
+    public static TreeSet<String> getAllKeywordsUsingRegex(File file) throws IOException {
+        Pattern tokenBlockPattern = Pattern.compile("TOKEN\\s*:\\s*(?:/\\*.*\\*/*)\\n\\{(?:[^\\}\\{]+|\\{(?:[^\\}\\{]+|\\{[^\\}\\{]*\\})*\\})*\\}", Pattern.MULTILINE);
+        Pattern tokenStringValuePattern = Pattern.compile("\\\"(\\w{2,})\\\"", Pattern.MULTILINE);
+
+        TreeSet<String> allKeywords = new TreeSet<>();
+
+        Path path = file.toPath();
+        Charset charset = Charset.defaultCharset();
+        String content = new String(Files.readAllBytes(path), charset);
+
+        Matcher tokenBlockmatcher = tokenBlockPattern.matcher(content);
+        while (tokenBlockmatcher.find()) {
+            String tokenBlock = tokenBlockmatcher.group(0);
+            Matcher tokenStringValueMatcher= tokenStringValuePattern.matcher(tokenBlock);
+            while (tokenStringValueMatcher.find()) {
+                String tokenValue=tokenStringValueMatcher.group(1);
+                // test if pure US-ASCII
+                if (CHARSET_ENCODER.canEncode(tokenValue) && tokenValue.matches("[A-Za-z]+")) {
+                    allKeywords.add(tokenValue);
+                }
+            }
+        }
+        return allKeywords;
+    }
+
+    private static void addTokenImage(TreeSet<String> allKeywords, RStringLiteral literal) {
+        if (CHARSET_ENCODER.canEncode(literal.image) && literal.image.matches("[A-Za-z]+")) {
+            allKeywords.add(literal.image);
+        }
+    }
+
+    @SuppressWarnings({"PMD.EmptyIfStmt", "PMD.CyclomaticComplexity"})
+    private static void addTokenImage(TreeSet<String> allKeywords, Object o) throws Exception {
+        if (o instanceof RStringLiteral) {
```

Comment:
This would benefit from:  https://www.baeldung.com/java-pattern-matching-instanceof. But I think it is limited to JDK8 for now.